content/blog/pulumi-convert-terraform-improvements/index.md

```diff
@@ -0,0 +1,533 @@
+))
+
+We plan on continuing to polish the java experience to remove manual steps like this.
+
+{{% /choosable %}}
+
+{{% choosable language yaml %}}
+
+```shell
+pulumi convert --from terraform --language yaml --out pulumi-yaml-program
+```
+
+I have opted to output it to a different directory to preserve the contents of my Terraform project.
+
+### Project structure
+
+If we navigate to the `pulumi-yaml-program` directory, we can see a few things:
+
+- Pulumi.yaml, the pulumi configuration file.
+- Main.yaml, our equivalent to main.tf for golang projects.
+- sdks directory, where the generated bridged provider is output.
+
+{{% /choosable %}}
+
+{{% /chooser %}}
+
+### Cleanup
+
+If we simply run `pulumi up` this project unfortunately won't run right away.
+We need to create our stack configuration that contains the GCP and PlanetScale credentials
+(you may have noticed some warnings to the tune of
+`warning: /Users/PulumiUser/src/MyAwesomeProject/deployment/main.tf:16,1-18: Failed to evaluate provider config; Could not evaluate expression for gcp:project`
+that indicate these config values need to be set up in Pulumi).
+
+These were part of our `main.tf` file `variables`, but the code generator doesn't
+distinguish between provider config and code, so we need to handle this
+part on our own (for now).
+
+These can go in either your `Pulumi.yaml` or your stack configuration.  Places
+for them are already generated, you just need to fill them in:
+
+```yaml
+name: terraform-convert-example
+runtime: nodejs
+config:
+  gcp:project:
+    value: 'TODO: var.gcp_project' # fill in here
+  gcp:region:
+    value: 'TODO: var.gcp_region'
+  gcp:zone:
+    value: 'TODO: var.gcp_zone'
+  planetscale:serviceToken:
+    value: 'TODO: var.planetscale_service_token'
+  planetscale:serviceTokenName:
+    value: planetscaletoken
+```
+
+After filling things in the `Pulumi.yaml` should look something like this:
+
+```yaml
+name: terraform-convert-example
+runtime: nodejs
+config:
+  gcp:project:
+    value: 'brandonpollack23'
+  gcp:region:
+    value: 'us-central1'
+  gcp:zone:
+    value: 'us-central1-a'
+  planetscale:serviceToken:
+    value: 'supersecrettoken'
+  planetscale:serviceTokenName:
+    value: planetscaletoken
+```
+
+{{% notes type="warning" %}}
+Once again-and it is worth reiterating-**never *ever*** store secrets in a git repository
+that is publicly visible. Even in a private repository, certain secrets might
+not be something everyone with access to the source in your repository should
+have access to.
+
+We highly recommend you use a secrets and configuration service like [Pulumi ESC](https://www.pulumi.com/docs/esc/)
+to handle this kind of thing.
+{{% /notes %}}
+
+Now if you run `pulumi preview` it successfully generates a plan, and if you
+used real credentials in your Pulumi.yaml it'll deploy with `pulumi up`.
+
+There is still more you can do, the generated code could be cleaned up a bit since there
```

Comment:
Suggest: "There is still more you can do! The [...]"

Personally don't like the repetition of "more that you can do!" at the start and end of this paragraph. Maybe simplify entire paragraph to "Once your infrastructure is converted to Pulumi there is so much _more_ that you can do." and move the "remove unused variables" to the starting suggestions section - simpler and flows into that next section?